meta-mender-variscite/0001-Changed-mrproper-execution-sanity-to-after-config-sa.patch

```diff
@@ -0,0 +1,28 @@
+From 763ca5f9506d56368170db33eeab12aaabe03d1e Mon Sep 17 00:00:00 2001
+From: Esteban AP <esteban.aguero@northern.tech>
+Date: Wed, 5 Apr 2023 20:03:48 -0600
+Subject: [PATCH] Changed mrproper execution (sanity) to after config sanity
+ check
+
+Signed-off-by: Esteban AP <esteban.aguero@northern.tech>
+---
+ meta-mender-core/recipes-bsp/u-boot/u-boot-mender.inc | 3 +++
+ 1 file changed, 3 insertions(+)
+
+diff --git a/meta-mender-core/recipes-bsp/u-boot/u-boot-mender.inc b/meta-mender-core/recipes-bsp/u-boot/u-boot-mender.inc
+index 07f7f076..4b5a78f1 100644
+--- a/meta-mender-core/recipes-bsp/u-boot/u-boot-mender.inc
++++ b/meta-mender-core/recipes-bsp/u-boot/u-boot-mender.inc
+@@ -266,6 +266,9 @@ do_configure:prepend:mender-uboot() {
+         fi
+     done
+     unset IFS
++
++    # For Variscite
```

Comment:
That sounds like a band-aid, not like a fix. What is the actual problem?

Reply:
The issue is, the build fail after the sanity test to the .config (u-boot) is performed (meta-mender/meta-mender-core/recipes-bsp/u-boot/u-boot-mender.inc:215). The specific log:
```
DEBUG: Executing shell function do_compile
NOTE: make -j 16 CROSS_COMPILE=aarch64-fslc-linux- CC=aarch64-fslc-linux-gcc  --sysroot=/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot V=1 HOSTCC=gcc  -isystem/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/usr/include -O2 -pipe -L/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/usr/lib                         -L/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/lib                         -Wl,--enable-new-dtags                         -Wl,-rpath-link,/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/usr/lib                         -Wl,-rpath-link,/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/lib                         -Wl,-rpath,/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/usr/lib                         -Wl,-rpath,/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/lib                         -Wl,-O1 -Wl,--allow-shlib-undefined -Wl,--dynamic-linker=/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/sysroots-uninative/x86_64-linux/lib/ld-linux-x86-64.so.2 STAGING_INCDIR=/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/usr/include STAGING_LIBDIR=/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/usr/lib -C /buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git O=/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/imx8mm_var_dart_config all
make: Entering directory '/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git'
make -C /buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/imx8mm_var_dart_config KBUILD_SRC=/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git \
-f /buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/Makefile all
make[1]: Entering directory '/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/imx8mm_var_dart_config'
make -f ../Makefile syncconfig
make -f ../scripts/Makefile.build obj=scripts/basic
ln -fsn .. source
/bin/bash ../scripts/mkmakefile ..
rm -f .tmp_quiet_recordmcount
  GEN     Makefile
make -f ../scripts/Makefile.build obj=scripts/kconfig syncconfig
scripts/kconfig/conf  --syncconfig Kconfig
make -f ../scripts/Makefile.autoconf || \
	{ rm -f include/config/auto.conf; false; }
mkdir -p include/asm
if [ -d /buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/arch/arm/mach-imx8m/include/mach ]; then	\
	dest=arch/arm/mach-imx8m/include/mach;			\
else									\
	dest=arch/arm/include/asm/arch-imx8m;	\
fi;									\
ln -fsn /buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/$dest include/asm/arch
set -e; mkdir -p include/; 	(echo "/* Automatically generated - do not edit */"; for i in $(echo "" | sed 's/,/ /g'); do echo \#define CONFIG_$i | sed '/=/ {s/=/	/;q; } ; { s/$/	1/; }'; done; echo \#define CONFIG_BOARDDIR board/variscite/imx8mm_var_dart; echo \#include \<config_uncmd_spl.h\>; echo \#include \<configs/"imx8mm_var_dart".h\>; echo \#include \<asm/config.h\>; echo \#include \<linux/kconfig.h\>; echo \#include \<config_fallbacks.h\>; echo \#include \<config_mender.h\>;) < ../scripts/Makefile.autoconf > include/config.h.tmp; if [ -r include/config.h ] && cmp -s include/config.h include/config.h.tmp; then rm -f include/config.h.tmp; else : '  UPD     include/config.h'; mv -f include/config.h.tmp include/config.h; fi
mkdir -p spl/
  aarch64-fslc-linux-gcc  --sysroot=/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot -E -Wall -Wstrict-prototypes -Wno-format-security -fno-builtin -ffreestanding -std=gnu11 -fshort-wchar -fno-strict-aliasing -fno-PIE -O2 -fno-stack-protector -fno-delete-null-pointer-checks -Wno-pointer-sign -Wno-stringop-truncation -Wno-zero-length-bounds -Wno-array-bounds -Wno-stringop-overflow -Wno-maybe-uninitialized -fmacro-prefix-map=../= -g -fstack-usage -Wno-format-nonliteral -Wno-address-of-packed-member -Wno-unused-but-set-variable -Werror=date-time -Wno-packed-not-aligned -D__KERNEL__ -D__UBOOT__ -D__ARM__          -fno-pic  -mstrict-align -ffunction-sections -fdata-sections -fno-common -ffixed-r9  -mgeneral-regs-only    -fno-common -ffixed-x18 -pipe -Iinclude  -I../include -I../arch/arm/include -include ../include/linux/kconfig.h -nostdinc -isystem /buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/usr/bin/aarch64-fslc-linux/../../lib/aarch64-fslc-linux/gcc/aarch64-fslc-linux/11.3.0/include  -DDO_DEPS_ONLY -dM ../include/common.h > u-boot.cfg.tmp && { grep 'define CONFIG_' u-boot.cfg.tmp | sed '/define CONFIG_IS_ENABLED(/d;/define CONFIG_IF_ENABLED_INT(/d;/define CONFIG_VAL(/d;' > u-boot.cfg; rm u-boot.cfg.tmp; } || { rm u-boot.cfg.tmp; false; }
  aarch64-fslc-linux-gcc  --sysroot=/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot -x c -DDO_DEPS_ONLY -M -MP -Wall -Wstrict-prototypes -Wno-format-security -fno-builtin -ffreestanding -std=gnu11 -fshort-wchar -fno-strict-aliasing -fno-PIE -O2 -fno-stack-protector -fno-delete-null-pointer-checks -Wno-pointer-sign -Wno-stringop-truncation -Wno-zero-length-bounds -Wno-array-bounds -Wno-stringop-overflow -Wno-maybe-uninitialized -fmacro-prefix-map=../= -g -fstack-usage -Wno-format-nonliteral -Wno-address-of-packed-member -Wno-unused-but-set-variable -Werror=date-time -Wno-packed-not-aligned -D__KERNEL__ -D__UBOOT__ -D__ARM__          -fno-pic  -mstrict-align -ffunction-sections -fdata-sections -fno-common -ffixed-r9  -mgeneral-regs-only    -fno-common -ffixed-x18 -pipe -Iinclude  -I../include -I../arch/arm/include -include ../include/linux/kconfig.h -nostdinc -isystem /buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/usr/bin/aarch64-fslc-linux/../../lib/aarch64-fslc-linux/gcc/aarch64-fslc-linux/11.3.0/include -MQ include/config/auto.conf ../include/common.h > include/autoconf.mk.dep || { rm include/autoconf.mk.dep; false; }
  aarch64-fslc-linux-gcc  --sysroot=/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot -E -Wall -Wstrict-prototypes -Wno-format-security -fno-builtin -ffreestanding -std=gnu11 -fshort-wchar -fno-strict-aliasing -fno-PIE -O2 -fno-stack-protector -fno-delete-null-pointer-checks -Wno-pointer-sign -Wno-stringop-truncation -Wno-zero-length-bounds -Wno-array-bounds -Wno-stringop-overflow -Wno-maybe-uninitialized -fmacro-prefix-map=../= -g -fstack-usage -Wno-format-nonliteral -Wno-address-of-packed-member -Wno-unused-but-set-variable -Werror=date-time -Wno-packed-not-aligned -D__KERNEL__ -D__UBOOT__ -D__ARM__          -fno-pic  -mstrict-align -ffunction-sections -fdata-sections -fno-common -ffixed-r9  -mgeneral-regs-only    -fno-common -ffixed-x18 -pipe -Iinclude  -I../include -I../arch/arm/include -include ../include/linux/kconfig.h -nostdinc -isystem /buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot-native/usr/bin/aarch64-fslc-linux/../../lib/aarch64-fslc-linux/gcc/aarch64-fslc-linux/11.3.0/include -DCONFIG_SPL_BUILD -DDO_DEPS_ONLY -dM ../include/common.h > spl/u-boot.cfg.tmp && { grep 'define CONFIG_' spl/u-boot.cfg.tmp | sed '/define CONFIG_IS_ENABLED(/d;/define CONFIG_IF_ENABLED_INT(/d;/define CONFIG_VAL(/d;' > spl/u-boot.cfg; rm spl/u-boot.cfg.tmp; } || { rm spl/u-boot.cfg.tmp; false; }
mkdir -p spl/include/
  sed -n -f ../tools/scripts/define2mk.sed u-boot.cfg | while read line; do if [ -n "" ] || ! grep -q "${line%=*}=" include/config/auto.conf; then echo "$line"; fi; done > include/autoconf.mk
  sed -n -f ../tools/scripts/define2mk.sed spl/u-boot.cfg | while read line; do if [ -n "" ] || ! grep -q "${line%=*}=" include/config/auto.conf; then echo "$line"; fi; done > spl/include/autoconf.mk
touch include/config/auto.conf
set -e; mkdir -p include/config/; 	echo "2022.04$(/bin/bash ../scripts/setlocalversion ..)" < include/config/auto.conf > include/config/uboot.release.tmp; if [ -r include/config/uboot.release ] && cmp -s include/config/uboot.release include/config/uboot.release.tmp; then rm -f include/config/uboot.release.tmp; else : '  UPD     include/config/uboot.release'; mv -f include/config/uboot.release.tmp include/config/uboot.release; fi
ln -fsn .. source
set -e; mkdir -p include/generated/; 	(if test -n "${SOURCE_DATE_EPOCH}"; then SOURCE_DATE="@${SOURCE_DATE_EPOCH}"; DATE=""; for date in gdate date.gnu date; do ${date} -u -d "${SOURCE_DATE}" >/dev/null 2>&1 && DATE="${date}"; done; if test -n "${DATE}"; then LC_ALL=C ${DATE} -u -d "${SOURCE_DATE}" +'#define U_BOOT_DATE "%b %d %C%y"'; LC_ALL=C ${DATE} -u -d "${SOURCE_DATE}" +'#define U_BOOT_TIME "%T"'; LC_ALL=C ${DATE} -u -d "${SOURCE_DATE}" +'#define U_BOOT_TZ "%z"'; LC_ALL=C ${DATE} -u -d "${SOURCE_DATE}" +'#define U_BOOT_EPOCH %s'; else return 42; fi; else LC_ALL=C date +'#define U_BOOT_DATE "%b %d %C%y"'; LC_ALL=C date +'#define U_BOOT_TIME "%T"'; LC_ALL=C date +'#define U_BOOT_TZ "%z"'; LC_ALL=C date +'#define U_BOOT_EPOCH %s'; fi) < ../Makefile > include/generated/timestamp_autogenerated.h.tmp; if [ -r include/generated/timestamp_autogenerated.h ] && cmp -s include/generated/timestamp_autogenerated.h include/generated/timestamp_autogenerated.h.tmp; then rm -f include/generated/timestamp_autogenerated.h.tmp; else : '  UPD     include/generated/timestamp_autogenerated.h'; mv -f include/generated/timestamp_autogenerated.h.tmp include/generated/timestamp_autogenerated.h; fi
/bin/bash ../scripts/mkmakefile ..
set -e; mkdir -p include/generated/; 	(if test -n "${DEVICE_TREE}"; then echo \#define DEVICE_TREE \"\"; else echo \#define DEVICE_TREE CONFIG_DEFAULT_DEVICE_TREE; fi) < ../Makefile > include/generated/dt.h.tmp; if [ -r include/generated/dt.h ] && cmp -s include/generated/dt.h include/generated/dt.h.tmp; then rm -f include/generated/dt.h.tmp; else : '  UPD     include/generated/dt.h'; mv -f include/generated/dt.h.tmp include/generated/dt.h; fi
  GEN     Makefile
make -f ../scripts/Makefile.build obj=scripts/basic
  if [ -f "FORCE" ]; then cat FORCE > include/generated/env.txt; elif [ -n "" ]; then echo "Missing file ../board/variscite/imx8mm_var_dart/.env"; else touch include/generated/env.txt ; fi
rm -f .tmp_quiet_recordmcount
if [ -f ../.config -o -d ../include/config ]; then \
	echo >&2 "  .. is not clean, please run 'make mrproper'"; \
	echo >&2 "  in the '..' directory.";\
	/bin/false; \
fi;
set -e; mkdir -p include/generated/; 	(echo \#define PLAIN_VERSION \"2022.04-lf_v2022.04_var01+g49ec7c516a\"; echo \#define U_BOOT_VERSION \"U-Boot \" PLAIN_VERSION; echo \#define U_BOOT_VERSION_NUM 2022; echo \#define U_BOOT_VERSION_NUM_PATCH $(echo 04 | sed -e "s/^0*//"); echo \#define CC_VERSION_STRING \"$(LC_ALL=C aarch64-fslc-linux-gcc  --sysroot=/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/recipe-sysroot --version | head -n 1)\"; echo \#define LD_VERSION_STRING \"$(LC_ALL=C aarch64-fslc-linux-ld.bfd --version | head -n 1)\"; ) < include/config/uboot.release > include/generated/version_autogenerated.h.tmp; if [ -r include/generated/version_autogenerated.h ] && cmp -s include/generated/version_autogenerated.h include/generated/version_autogenerated.h.tmp; then rm -f include/generated/version_autogenerated.h.tmp; else : '  UPD     include/generated/version_autogenerated.h'; mv -f include/generated/version_autogenerated.h.tmp include/generated/version_autogenerated.h; fi
  .. is not clean, please run 'make mrproper'
  in the '..' directory.
make[1]: *** [/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/Makefile:1928: prepare3] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/imx8mm_var_dart_config'
make: *** [Makefile:177: sub-make] Error 2
make: Leaving directory '/buildah/estape/nt_projects/variscite-imx8mm-kirkstone-imx-community/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git'
ERROR: oe_runmake failed
WARNING: exit code 1 from a shell command.
```
Apparently the cleandir part (http://cgit.openembedded.org/openembedded-core/commit/meta/recipes-bsp/u-boot?h=master-next&id=26023b6b0f897842fd98b3e10a8acd5b3ad8f418 vs https://github.com/varigit/meta-variscite-bsp/blob/kirkstone-var02/recipes-bsp/u-boot/u-boot-common.inc ), is missing from the uboot fork used in variscite environment. 
```
+B = "${WORKDIR}/build"
+do_configure[cleandirs] = "${B}"
```
I added it to the bappend in the commit 10e1a6962ce7ddd25603e9d7280586974660f6f6